src/main/java/net/cardosi/mojo/cache/CachedProject.java

```diff
@@ -97,10 +106,10 @@ public void replace(Artifact artifact, MavenProject currentProject, List<CachedP
      *
      * TODO this could be updated to compare before/after hash and avoid marking children as dirty
      */
-    public void markDirty() {
+    public CompletableFuture<Void> markDirty() {
         synchronized (steps) {
             if (steps.isEmpty()) {
-                return;
+                return null;
```

Comment:
Maybe its in a pending patch, but you don't seem to use the CF result here at all? and "return null" two of the three sites seems wonky - perhaps just return CompletedFuture.completedFuture(null) at each of those, or revert the sig change if not needed?

Reply:
the result is never used, I was just trying to keep the code compiling. I'll change it to CompletedFuture.completedFuture(null)